rest-api/README.md

```diff
@@ -0,0 +1,321 @@
+| 409    | RFQ already exists     |
+| 409    | Thread already exists  |
+
+## Get Quote
+### Endpoint
+`GET /threads/:thread_id/?messageType=quote`
+
+### Response
+| Status             | Body                              |
+| ------------------ | --------------------------------- |
+| `200: OK`          | `{ data: TbdexMessage<Quote>[] }` |
+| `400: Bad Request` | `{ errors: Error[] }`             |
+
+### Response Body
+> [!IMPORTANT]
+> See Quote structure [here](../README.md#quote)
+
+## Submit Order
+
+### Endpoint
+`POST /threads/:thread_id/order`
+
+### Order Request Body
+> [!IMPORTANT]
+> See Order structure [here](../README.md#order)
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description                                                             |
+| ------ | ----------------------------------------------------------------------- |
+| 400    | Failed Signature Check                                                  |
+| 404    | Thread not found                                                        |
+| 409    | Order already exists                                                    |
+| 409    | Order not allowed (e.g. bc thread was closed or bc no quote exists yet) |
+
+## Submit Close
+
+### Description
+Closes the thread. Indicates that Alice is no longer interested
+
+### Endpoint
+`POST /threads/:thread_id/close`
+
+### Request Body
+> [!IMPORTANT]
+> See Close structure [here](../README.md#close)
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description            |
+| ------ | ---------------------- |
+| 400    | Failed Signature Check |
+| 404    | Thread not found       |
+| 409    | Close not allowed      |
+
+---
+
+## Get Thread
+
+### Description
+Retrieves the thread specified by ID and messageType
+
+### Authentication
+N/A
+
+### Endpoint
+`GET /threads/:id`
+
+
+### Query Params
+| Param         | Description                   |
+| ------------- | ----------------------------- |
+| `messageType` | filters the messages returned |
+
+
+### Response
+
+| Status             | Body                       |
+| ------------------ | -------------------------- |
+| `200: OK`          | `{ data: TbdexMessage[] }` |
+| `400: Bad Request` | `{ errors: Error[] }`      |
+| `404: Not Found`   | N/A                        |
+
+---
+
+## List Threads
+
+### Description
+Returns an array containing IDs of threads started by the requesting DID
+
+### Endpoint
+`GET /threads`
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `200: OK.     `    | `{ data: string[]}`   |
```

Comment:
thoughts on returning TbDEXMessage[][] - so we get a list of exchange, and each exchange consists of a list of tbdexmessages? 

we are thinking that having a didpay endpoint that retrieves all exchanges would be nice so we can render a list of exchanges in the 'landing page'

Reply:
and also a query param so we can pass in only the exchangeIds that we need?